docs/src/main/asciidoc/building-my-first-extension-guide.adoc

```diff
@@ -0,0 +1,700 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Building my first extension
+
+include::./attributes.adoc[]
+
+https://quarkus.io/extensions/[Quarkus extensions] enhance your application just as projects dependencies do.
+The role of the extensions is about leveraging Quarkus paradigms to integrate seamlessly a library into Quarkus.
+This is how you can use your battle-tested ecosystem and take advantage of Quarkus performance and native compilation.
+
+In this guide we are going to develop the *Sample Greeting Extension*.
+The extension will expose a customizable HTTP endpoint which simply greets the visitor.
```

Comment:
I think it should be extra clear that:
* you don't need an extension to add a servlet
* it's a simplified example so you don't really see the power of moving things to build time or simplifying the build of native images